### PR TITLE
docs: Docs improvements

### PR DESCRIPTION
**Pull request by jhmarina**
# Describe your changes

- Adds a README file explaining how to set up & generate the documentation locally
- Sorts out the issue in the animation by only having 1 hierarchy level in the menu. Trying to have 2 levels breaks the UX.
- Merges some documents, which is needed to sort out the issue above ☝️
- Adds missing ADRs. I've had to move the ADRs outside of the "Development manual" section due to the broken UX

## Issue ticket number and link
Closes https://github.com/orgs/aquarist-labs/projects/5/views/12

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

**Comment by giubacc** [comment]
There are a lot of changes, I had a look, it looks sound.
It seems that you must rebase this PR on the current main btw.

File: docs/developing.md [hunk 1]
```diff
@@ -89,3 +89,225 @@ against the running RGW. E.g., `s3cmd mb s3://foo`, to create a new bucket.
 [2]: https://github.com/aquarist-labs/ceph/tree/s3gw
 [3]: https://docs.ceph.com/en/pacific/install/build-ceph/#id1
 [4]: https://github.com/s3tools/s3cmd
+
+## How to build your own containers
```
Comment (votdev) [inline docs/developing.md:93]: This section (line 93-313) is a duplication of the `build-instructions.md`. Would it make sense to link to this file here?

**Review by votdev — CHANGES_REQUESTED** [review]

File: docs/developing.md (hunk 1 above)
Comment (jhmarina) [reply docs/developing.md:93]: The idea is to concentrate both instructions in this document, which sorts out the UX issue. We could have potentially just added links to the files instead but I wanted to save the user that extra click. I just deleted `build-ui.md`

**Review by votdev — DISMISSED** [review]

**Review by giubacc — DISMISSED** [review]
lgtm

**Comment by giubacc** [comment]
please, sign your commits

**Comment by jhmarina** [comment]
> please, sign your commits

damn, I didn't have the gpg key set up properly again. It should be good now. Mind reaproving?

**Review by giubacc — APPROVED** [review]

**Pull request merged**